spec/concepts/stock/task/load_spec.rb

```diff
@@ -41,24 +41,55 @@
     end
   end
 
-  context 'when new stock is not importable' do
-    let!(:current_stock) { create :stock, :of_july, :completed }
-    let(:remote_stock)   { build :stock, :of_july, :pending }
+  context 'when remete stock is not importable' do
+    context 'because current stock completed' do
+      let!(:current_stock) { create :stock, :of_july, :completed }
+      let(:remote_stock)   { build :stock, :of_july, :pending }
 
-    it { is_expected.to be_failure }
+      it { is_expected.to be_success }
 
-    it 'logs a warning' do
-      subject
-      expect(logger)
-        .to have_received(:warn)
-        .with('Remote stock not importable (remote month: 07, current (COMPLETED) month: 07)')
+      it 'logs a warning' do
+        subject
+        expect(logger)
+          .to have_received(:warn)
+          .with('Remote stock not importable (remote month: 07, current (COMPLETED) month: 07)')
+      end
+
+      its([:remote_stock]) { is_expected.not_to be_persisted }
+
+      it 'does not enqueue job' do
+        expect { subject }
+          .not_to have_enqueued_job ImportStockJob
+      end
     end
 
-    its([:remote_stock]) { is_expected.not_to be_persisted }
+    context 'because current stock is stuck' do
+      let!(:current_stock) { create :stock, :of_july, status: 'LOADING' }
+      let(:remote_stock)   { build :stock, :of_july, :pending }
 
-    it 'does not enqueue job' do
-      expect { subject }
-        .not_to have_enqueued_job ImportStockJob
+      it { is_expected.to be_failure }
+
+      it 'logs a warning' do
+        subject
+        expect(logger)
+          .to have_received(:warn)
+          .with('Remote stock not importable (remote month: 07, current (LOADING) month: 07)')
+      end
+
+      it 'logs an error' do
+        subject
+        expect(logger)
+          .to have_received(:error)
+          .with('Current stock is stuck in LOADING')
```

Comment:
LOADING ça veut dire en cours ? Si oui pareil, "Current stock is still in progress" c'est plus clair. Pareil pour les différents messages de logs en fait.

Reply:
idem que pour l'autre message, je trouve qu'il est plus important d'avoir dans les logs le vrai status qu'un message clair, du coup je vais mettre les deux aussi ici :
"Current stock is still in progress (LOADING)"

Reply:
Ok, tant que les messages d'erreurs sont plus clair ça me va.

Reply:
mise à jour